libdivide.h

```diff
@@ -18,9 +18,51 @@
 #include <stdint.h>
 
 #if defined(__cplusplus)
+#if __AVR__
+
+// AVR doesn't ship with any of the C++ standard library. So manually
+// define traits for a few integer types.
+namespace std {
+
+template<typename _Tp, _Tp __v>
+    struct integral_constant
+    {
+      static constexpr _Tp                  value = __v;
```

Comment:
`constexpr` is a C++11 thing, we try to stay C++98 compatible, please make this just `const`.

Alternatively, all these checks are sort of silly, we should remove the `IS_INTEGRAL` etc. checks in `dispatcher` and just specialize with our four types (`uint32`, etc). You can do that if you like, otherwise what you have here is fine.

Reply:
> Alternatively, all these checks are sort of silly, we should remove the `IS_INTEGRAL` etc. checks in `dispatcher` and just specialize with our four types (`uint32`, etc). You can do that if you like, otherwise what you have here is fine.

Done! Code is much simpler now.